app/src/main/java/com/google/samples/apps/sunflower/adapters/SunflowerPagerAdapter.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.adapters
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import com.google.samples.apps.sunflower.GardenFragment
+import com.google.samples.apps.sunflower.PlantListFragment
+import kotlin.IndexOutOfBoundsException
+
+const val MY_GARDEN_PAGE_INDEX = 0
+const val PLANT_LIST_PAGE_INDEX = 1
+
+class SunflowerPagerAdapter(
+    private val tabTitles: HashMap<Int, String>,
```

Comment:
prefer *SparseArrayCompat*

Reply:
For readability, we prefer to use `HashMap` here.